services/web/server/src/simcore_service_webserver/users/_users_repository.py

```diff
@@ -0,0 +1,422 @@
+import contextlib
+from typing import Any
+
+import simcore_postgres_database.errors as db_errors
+import sqlalchemy as sa
+from aiohttp import web
+from common_library.users_enums import UserRole
+from models_library.groups import GroupID
+from models_library.users import (
+    MyProfile,
+    UserBillingDetails,
+    UserID,
+    UserNameID,
+    UserPermission,
+)
+from pydantic import TypeAdapter, ValidationError
+from simcore_postgres_database.models.groups import groups, user_to_groups
+from simcore_postgres_database.models.products import products
+from simcore_postgres_database.models.users import UserStatus, users
+from simcore_postgres_database.models.users_details import (
+    users_pre_registration_details,
+)
+from simcore_postgres_database.utils_groups_extra_properties import (
+    GroupExtraPropertiesNotFoundError,
+    GroupExtraPropertiesRepo,
+)
+from simcore_postgres_database.utils_repos import (
+    pass_or_acquire_connection,
+    transaction_context,
+)
+from simcore_postgres_database.utils_users import (
+    UsersRepo,
+    generate_alternative_username,
+)
+from sqlalchemy import delete
+from sqlalchemy.engine.row import Row
+from sqlalchemy.ext.asyncio import AsyncConnection, AsyncEngine
+
+from ..db.plugin import get_asyncpg_engine
+from ._common.models import FullNameDict, ToUserUpdateDB
+from .exceptions import (
+    BillingDetailsNotFoundError,
+    UserNameDuplicateError,
+    UserNotFoundError,
+)
+
+_ALL = None
+
+
+def _parse_as_user(user_id: Any) -> UserID:
+    try:
+        return TypeAdapter(UserID).validate_python(user_id)
+    except ValidationError as err:
+        raise UserNotFoundError(uid=user_id, user_id=user_id) from err
+
+
+async def get_user_or_raise(
+    engine: AsyncEngine,
+    connection: AsyncConnection | None = None,
+    *,
+    user_id: UserID,
+    return_column_names: list[str] | None = _ALL,
+) -> dict[str, Any]:
+    if return_column_names == _ALL:
+        return_column_names = list(users.columns.keys())
+
+    assert return_column_names is not None  # nosec
+    assert set(return_column_names).issubset(users.columns.keys())  # nosec
+
+    async with pass_or_acquire_connection(engine, connection) as conn:
+        result = await conn.stream(
+            sa.select(*(users.columns[name] for name in return_column_names)).where(
```

Comment:
what happens if I pass `[]` ?

Reply:
good point. I would say that the current code would returns None, but it has been a waste to transmit that ...